### ISSUE TITLE
Reviewing test_000013 - test_000035

### ISSUE THREAD

**Issue by yilun-zhangs** [AOMediaCodec/libiamf#13]
Hello~

**Common Issue(Opus codec):**
Issue 1:  preskip value is 0 in codec_config_metadata { }
Refering to spec, [Pre-skip](https://tools.ietf.org/html/rfc7845#) shall be same as the number of audio samples to be trimmed at the start of substreams.

Issue 2: relative_offset of OBU_DATA_TYPE_PARAMETER is 0
Refering to https://github.com/AOMediaCodec/iamf/pull/331, In my opinion, the relative_offset of parameter is same with the value of trimming start.

Issue 3: no edit list box in mp4 file.
Refering to spec, Take the number of trimmed samples from the IA sequence and converting it to the sample rate used by the [elst](https://www.iso.org/standard/68960.html#) boxes.

Issue 4: After decoding, the wave quality is poor comparing with original wav file.
I think, this could be resolved by improving the bitrate setting, I guess you have set with a low bitrate.

**Individual Issues:**
Issue 5: test_000015, the parameter_id in parameter_block_metadata{} is 100 and 101, but in element_mix_config{} and element_mix_config{}, both parameter IDs are 100, not matching.

Thanks.

**Comment by jwcullen** [comment]
All of test_000013 - test_000035 uses `samples_to_trim_at_start == 0`. 
* Issue 1: Therefore preskip value in Opus is 0
* Issue 2: Therefore the `relative_offset` of the Parameter block is 0
* Issue 3: We left out the edit list when it is zero as well.
* But maybe it Opus should always have samples to trim at the start?

Issue 4: I think this is in part because they are very short artificial samples. 

Issue 5: Please note that test_000015 has `is_valid = false`. The problem you describe was intention. We made an invalid test vector that had an extra Parameter Block that is not used by anything else.

**Comment by jwcullen** [comment]
Can we track further issues to the initial drop as either new issues or on this issue and get the outstanding PRs accepted and merged in?

Because there are a few outstanding PRs that depend on each other, so I'm worried it is causing some confusion.

**Comment by yilun-zhangs** [comment]
> 

Hello~

Regarding one of the opus TCs, test_000020, after decoding, wecan find the beginning wave is not aligned with original file.
It means there is preskip when do encoding. So I think at encoder side, you may need to get the preskip value from opus encoder(If libopus is used, you can call opus_multistream_encoder_ctl(, OPUS_GET_LOOKAHEAD(value))).

Thanks.

**Comment by yilun-zhangs** [comment]
> Can we track further issues to the initial drop as either new issues or on this issue and get the outstanding PRs accepted and merged in?
> 
> Because there are a few outstanding PRs that depend on each other, so I'm worried it is causing some confusion.

Yes, there are too many PRs which may make confusion. How about accepting the PRs and merging in firstly. To new issues, then create new PR to fix?

**Comment by kr728-kim** [comment]
@jwcullen In test_000013, `substream_count` and `coupled_substream_count` are 2 and 0 each, although the layout is stereo.

**Comment by yeroro** [comment]
> @jwcullen In test_000013, `substream_count` and `coupled_substream_count` are 2 and 0 each, although the layout is stereo.

Again, you are looking at an outdated file. I will merge them all.

**Comment by jwcullen** [comment]
We have addressed everything here. Other subsequent issues are covered up other open follow up issues. I will close this in a few days if there are no further comments.
